File: Java Project/Iteration_2/src/CompulsoryCourse.java
package Iteration_2.src;



import java.util.ArrayList;
import java.util.HashMap;
import java.util.Random;

public  class CompulsoryCourse {

	private String courseCode;
	private  String courseName;
	private int credit;
	private ArrayList<String> prerequisites; //includes courseCodes only
	private Lecturer lecturer;
	private  CourseSection courseSection;
	private ArrayList<Student> courseStudents;
	private  String openedTerm;


	//Constructor with parameters takes in a course code, course name, credit, prerequisites, schedule, semester and course students


	public CompulsoryCourse(String courseCode, String courseName, int credit, ArrayList<String> prerequisites, Lecturer lecturer, CourseSection courseSection, ArrayList<Student> courseStudents, String givenSemester) {
		this.courseCode = courseCode;
		this.courseName = courseName;
		this.credit = credit;
		this.prerequisites = prerequisites;
		this.lecturer = lecturer;
		this.courseSection = courseSection;
		this.courseStudents = courseStudents;
		this.openedTerm = givenSemester;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public String getCourseCode() {
		return courseCode;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setCourseCode(String courseCode) {
		this.courseCode = courseCode;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public String getCourseName() {
		return courseName;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setCourseName(String courseName) {
		this.courseName = courseName;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public int getCredit() {
		return credit;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setCredit(int credit) {
		this.credit = credit;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setPrerequisites(ArrayList<String> prerequisites) {
		this.prerequisites = prerequisites;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setLecturer(Lecturer lecturer) {
		this.lecturer = lecturer;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public String getOpenedTerm() {
		return openedTerm;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setOpenedTerm(String openedTerm) {
		this.openedTerm = openedTerm;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public ArrayList<String> getPrerequisites() {
		return prerequisites;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}
	//Getter for semester property
	//Getter for schedule property
	public Lecturer getLecturer() {
		return lecturer;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}
	//Getter for course students property
	public ArrayList<Student> getCourseStudents() {
		return courseStudents;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}
	//Setter for course students property
	public void setCourseStudents(ArrayList<Student> courseStudent) {
		this.courseStudents = courseStudents;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public CourseSection getCourseSection() {
		return courseSection;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setCourseSection(CourseSection courseSection) {
		this.courseSection = courseSection;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}
	public Schedule getRandomSection(){
		int size =this.getCourseSection().getSchedule().size();
		HashMap<String,Schedule> scheduleHashMap = this.getCourseSection().getSchedule();
		Random random = new Random();
		return size == 1 ? scheduleHashMap.get(String.valueOf(1)):scheduleHashMap.get(String.valueOf(random.nextInt(1,size)));
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();

	}

	@Override
	public String toString() {
		return "CompulsoryCourse{" +
				"courseCode='" + courseCode + '\'' +
				", courseName='" + courseName + '\'' +
				", credit=" + credit +
				", prerequisites=" + prerequisites +
				", lecturer=" + lecturer +
				", courseSection=" + courseSection +
				", courseStudents=" + courseStudents +
				", givenSemester='" + openedTerm + '\'' +
				'}';
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}
}
class CourseSection {

	private String courseCode;
	private int sectionCount;
	private HashMap<String,Schedule> schedule;
	private int quota;

	public CourseSection(String courseCode, int sectionCount, HashMap<String, Schedule> schedule, int quota) {
		this.courseCode = courseCode;
		this.sectionCount = sectionCount;
		this.schedule = schedule;
		this.quota = quota;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public CourseSection() {
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public String getCourse() {
		return courseCode;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setCourse(String courseCode) {
		this.courseCode = courseCode;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public int getSectionID() {
		return sectionCount;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setSectionCount(int sectionCount) {
		this.sectionCount = sectionCount;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}


	public String getCourseCode() {
		return courseCode;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setCourseCode(String courseCode) {
		this.courseCode = courseCode;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public int getSectionCount() {
		return sectionCount;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public HashMap<String, Schedule> getSchedule() {
		return schedule;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setSchedule(HashMap<String, Schedule> schedule) {
		this.schedule = schedule;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public int getQuota() {
		return quota;
		Iteration_2.src.LoggingAgent LoggingAgent = new Iteration_2.src.LoggingAgent();
        	LoggingAgent.LoggerMethod();
	}

	public void setQuota(int quota) {
		this.quota = quota;
	}


	@Override
	public String toString() {
		return "CourseSection{" +
				"courseCode='" + courseCode + '\'' +
				", sectionCount=" + sectionCount +
				", schedule=" + schedule +
				", quota=" + quota +
				'}';
	}
}

File: Python/src/CompulsoryCourse.py
import random

class CompulsoryCourse:

	#Constructor with parameters takes in a course code, course name, credit, prerequisites, schedule, semester and course students


	def __init__(self, courseCode, courseName, credit, prerequisites, lecturer, courseSection, courseStudents, givenSemester):
		self.__courseCode = courseCode
		self.__courseName = courseName
		self.__credit = credit
		self.__prerequisites = prerequisites
		self.__lecturer = lecturer
		self.__courseSection = courseSection
		self.__courseStudents = courseStudents
		self.__openedTerm = givenSemester

	def getCourseCode(self):
		return self.__courseCode


	def setCourseCode(self, courseCode):
		self.__courseCode = courseCode


	def getCourseName(self):
		return self.__courseName


	def setCourseName(self, courseName):
		self.__courseName = courseName


	def getCredit(self):
		return self.__credit


	def setCredit(self, credit):
		self.__credit = credit


	def setPrerequisites(self, prerequisites):
		self.__prerequisites = prerequisites


	def setLecturer(self, lecturer):
		self.__lecturer = lecturer


	def getOpenedTerm(self):
		return self.__openedTerm


	def setOpenedTerm(self, openedTerm):
		self.__openedTerm = openedTerm


	def getPrerequisites(self):
		return self.__prerequisites

	#Getter for semester property
	#Getter for schedule property
	def getLecturer(self):
		return self.__lecturer

	#Getter for course students property
	def getCourseStudents(self):
		return self.__courseStudents

	#Setter for course students property
	def setCourseStudents(self, courseStudents):
		self.__courseStudents = courseStudents


	def getCourseSection(self):
		return self.__courseSection


	def setCourseSection(self, courseSection):
		self.__courseSection = courseSection

	def getRandomSection(self):
		size = self.getCourseSection().getSchedule().size()
		scheduleHashMap = self.getCourseSection().getSchedule()
		r1 = random.randint(1,size)
		return scheduleHashMap.get(1) if size == 1 else scheduleHashMap.get(r1)


	def toString(self):
		return "CompulsoryCourse{" + "courseCode='" + self.__courseCode + '\'' + ", courseName='" + self.__courseName + '\'' + ", credit=" + self.__credit + ", prerequisites=" + self.__prerequisites + ", lecturer=" + self.__lecturer + ", courseSection=" + self.__courseSection + ", courseStudents=" + self.__courseStudents + ", givenSemester='" + self.__openedTerm + '\'' + '}'


class CourseSection:

	#courseCode = ""
	#sectionCount = 0
	#schedule
	#quota = 0

	def __init__(self, courseCode, sectionCount, schedule, quota):
		self.__courseCode = courseCode
		self.__sectionCount = sectionCount
		self.__schedule = schedule
		self.__quota = quota

	def getCourse(self):
		return self.__courseCode


	def setCourse(self, courseCode):
		self.__courseCode = courseCode


	def getSectionID(self):
		return self.__sectionCount


	def setSectionCount(self, sectionCount):
		self.__sectionCount = sectionCount


	def getCourseCode(self):
		return self.__courseCode


	def setCourseCode(self, courseCode):
		self.__courseCode = courseCode


	def getSectionCount(self):
		return self.__sectionCount


	def getSchedule(self):
		return self.__schedule


	def setSchedule(self, schedule):
		self.__schedule = schedule


	def getQuota(self):
		return self.__quota


	def setQuota(self, quota):
		self.__quota = quota


	def toString(self):
		return "CourseSection{" + "courseCode='" + self.__courseCode + '\'' + ", sectionCount=" + self.__sectionCount + ", schedule=" + self.__schedule + ", quota=" + self.__quota + '}'
